core/state/shared_pool.go

```diff
@@ -0,0 +1,36 @@
+package state
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"sync"
+)
+
+// sharedStorage is used to store maps of originStorage of stateObjects
+type SharedStorage struct {
+	*sync.RWMutex
+	sharedMap map[common.Address]*sync.Map
+}
+
+func NewSharedStorage() *SharedStorage {
+	sharedMap := make(map[common.Address]*sync.Map, 1500)
+	return &SharedStorage{
+		&sync.RWMutex{},
+		sharedMap,
+	}
+}
+
+// Check whether the storage exist in pool,
+// new one if not exist, it will be fetched in stateObjects.GetCommittedState()
+func (storage *SharedStorage) getOrInertStorage(address common.Address) *sync.Map {
```

Comment:
`storage` ==> `s`

Reply:
I thin we can rename `getOrInertStorage` to `getStorage`, `insert` is an internal action that can be hide from caller.

Reply:
It is a bit weird  `SharedStorage` get a function called `getOrInertStorage`, how can we get storage from a storage, I will suggest rename `SharedStorage` to `StoragePool`

Reply:
sure